File: spine-corona/spine/SkeletonData.lua

local SkeletonData = {}
function SkeletonData.new (attachmentResolver)
	if not attachmentResolver then error("attachmentResolver cannot be nil", 2) end

	local self = {
		attachmentResolver = attachmentResolver,
		bones = {},
		slots = {},
		skins = {}
	}

	function self:findBone (boneName)
		if not boneName then error("boneName cannot be nil.", 2) end
		for i,bone in ipairs(self.bones) do
			if bone.name == boneName then return bone end
		end
		return nil
	end

	function self:findBoneIndex (boneName)
		if not boneName then error("boneName cannot be nil.", 2) end
		for i,bone in ipairs(self.bones) do
			if bone.name == boneName then return i end
		end
		return -1
	end

	function self:findSlot (slotName)
		if not slotName then error("slotName cannot be nil.", 2) end
		for i,slot in ipairs(self.slots) do
			if slot.name == slotName then return slot end
		end
		return nil
	end

	function self:findSlotIndex (slotName)
		if not slotName then error("slotName cannot be nil.", 2) end
		for i,slot in ipairs(self.slots) do
			if slot.name == slotName then return i end
		end
		return -1
	end

	function self:findSkin (skinName)
		if not skinName then error("skinName cannot be nil.", 2) end
		for i,skin in ipairs(self.skins) do
			if skin.name == skinName then return skin end
		end
		return nil
	end

	return self
end
return SkeletonData

File: spine-c/src/spine/SkeletonData.c
#include <spine/SkeletonData.h>
#include <spine/util.h>

SkeletonData* SkeletonData_create () {
	SkeletonData* self = CALLOC(SkeletonData, 1)
	return self;
}

void SkeletonData_dispose (SkeletonData* self) {
	FREE(self)
}

BoneData* SkeletonData_findBone (const SkeletonData* self, const char* boneName) {
	int i;
	for (i = 0; i < self->boneCount; ++i)
		if (strcmp(self->bones[i]->name, boneName) == 0) return self->bones[i];
	return 0;
}

int SkeletonData_findBoneIndex (const SkeletonData* self, const char* boneName) {
	int i;
	for (i = 0; i < self->boneCount; ++i)
		if (strcmp(self->bones[i]->name, boneName) == 0) return i;
	return 0;
}

SlotData* SkeletonData_findSlot (const SkeletonData* self, const char* slotName) {
	int i;
	for (i = 0; i < self->slotCount; ++i)
		if (strcmp(self->slots[i]->name, slotName) == 0) return self->slots[i];
	return 0;
}

int SkeletonData_findSlotIndex (const SkeletonData* self, const char* slotName) {
	int i;
	for (i = 0; i < self->slotCount; ++i)
		if (strcmp(self->slots[i]->name, slotName) == 0) return i;
	return 0;
}

Skin* SkeletonData_findSkin (const SkeletonData* self, const char* skinName) {
	int i;
	for (i = 0; i < self->skinCount; ++i)
		if (strcmp(self->skins[i]->name, skinName) == 0) return self->skins[i];
	return 0;
}
